rando_modules/modify_itempool.py

```diff
@@ -1,320 +1,129 @@
-                0x08E, # UltraShroom
-                0x0A2, # JamminJelly
-            ],
-            TYPE_TAYCETITEM: [
-                0x0B2, # HoneyUltra
-                0x0DA, # JellyShroom1
-            ],
-        },
-        9: {
-            TYPE_BATTLEITEM: [
-
-            ],
-            TYPE_HEALINGITEM: [
-
-            ],
-            TYPE_TAYCETITEM: [
-                0x0B3, # MapleUltra
-                0x0C6, # JellySuper
-                0x0D1, # JellyPop
-            ],
-        },
-        10: {
-            TYPE_BATTLEITEM: [
-                0x097, # RepelGel
-            ],
-            TYPE_HEALINGITEM: [
-
-            ],
-            TYPE_TAYCETITEM: [
-                0x0B4, # JellyUltra
-                0x0BE, # DeluxeFeast
-            ],
-        },
-    }
-
-    # item scarcity:
-    # when placing a non-essential item, have a chance to replace the item with
-    # another from a lower item tier according to...
-    #
-    # 0    vanilla
-    # 1    30% chance to divide tier by 2
-    # 2    40% chance to divide tier by 2, 10% chance to divide tier by 3
-    # 3    50% chance to divide tier by 2, 20% chance to divide tier by 3
-    # 4    60% chance to divide tier by 2, 40% chance to divide tier by 3
-    # 5    100% chance to divide tier by 3
-    #
-    # Also choose an item by type:
-    # battle items    40% chance
-    # healing items   35% chance
-    # Tayce T. items  25% chance
-    #
-    # If no item of the chosen type exists in the chosen tier, retry with lower
-    # tier. If tier 0 is reached this way, pick randomly
-
-    # setting: {chance in %: divider}
-    scarcity_factors = {
-        0: {100: 1},
-        1: { 70: 1,
-             30: 2},
-        2: { 50: 1,
-             40: 2,
-             10: 3},
-        3: { 30: 1,
-             50: 2,
-             20: 3},
-        4: { 60: 2,
-             40: 3},
-        5: {100: 3}
-    }
-
-    # item type: {chance in %}
-    itemtype_chances = {
-        TYPE_BATTLEITEM: 40,
-        TYPE_HEALINGITEM: 35,
-        TYPE_TAYCETITEM: 25
+    weights = {
+        "battle": 50,
+        "heal": 45,
+        "taycet": 5,
     }
+    item_weights = [weights[item["type"]] for item in item_scores]
 
-    new_itempool = []
-
-    for item_obj in itempool:
-        item_id = item_obj.value
-        scarcity_tier = -1
+    new_items = []
+    for _ in range(n):
+        item, = random.choices(item_scores, item_weights) 
+        new_items.append(item)
 
-        # Fetch scarcity tier of current item
-        for tier, type_dict in scarcity_tiers.items():
-            if isinstance(type_dict, int) and tier == 0 and item_id == type_dict:
-                scarcity_tier = tier
-                break
-            if isinstance(type_dict, dict):
-                for item_list in type_dict.values():
-                    if item_id in item_list:
-                        scarcity_tier = tier
-                        break
+    return new_items
```

Comment:
`return random.choices(item_scores, item_weights, k=n)` maybe? Would avoid the odd style of using `item, = [...]`

Reply:
Good call, I think this is a holdover from an iteration where more happened during this loop than simply appending the items to a list.